spec/03-plugins/36-request-transformer/02-access_spec.lua

```diff
@@ -2227,4 +2227,400 @@ describe("Plugin: request-transformer(access) [#" .. strategy .. "]", function()
     end)
   end)
 end)
+
+describe("Plugin: request-transformer(access) [#" .. strategy .. "] untrusted_lua=off", function()
+  local client
+
+  lazy_setup(function()
+    local bp = helpers.get_db_utils(strategy, {
+      "routes",
+      "services",
+      "plugins",
+    })
+
+    local route1 = bp.routes:insert({
+      hosts = { "testLuaOff.test" }
+    })
+
+    bp.plugins:insert {
+      route = { id = route1.id },
+      name = "request-transformer",
+      config = {
+        add = {
+          headers = {"x-added:$(query_params.user)"},
+        }
+      }
+    }
+
+    assert(helpers.start_kong({
+      database = strategy,
+      untrusted_lua = "off",
+      plugins = "bundled, request-transformer",
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+    }))
+  end)
+
+  lazy_teardown(function()
+    helpers.stop_kong()
+  end)
+
+  before_each(function()
+    client = helpers.proxy_client()
+  end)
+
+  after_each(function()
+    if client then client:close() end
+  end)
+
+  describe("using template", function()
```

Comment:
Should we also test use of expressions when `untrusted_lua=off` to make sure it does not work?

Reply:
I added a better unhappy path to the test, thanks @dndx .

Currently the behaviour of `untrusted_lua=off` is to just not render the template which will be treated as just a regular string; This allows us to not do any parsing of the template to determine whether it contains inline Lua, and [just disable it](https://github.com/Kong/kong/pull/10327/files#diff-41ae27766e102f74d9deb4899fc8ba265b0dcd602e2ed8a0a150d5fc03ab406fR47) on penlight. If we wanted to actually fail and return some error when Lua is detected though, we'd need to do some pattern matching, I'm still unsure which is the best approach.